pkg/tracing/tracing.go

```diff
@@ -20,113 +20,354 @@ import (
+		}
+	}
+	if s.enhancedSpan != nil {
+		if err != nil {
+			s.enhancedSpan.SetStatus(enhanced.StatusCodeError, err.Error())
+			s.enhancedSpan.RecordError(err)
+		} else {
+			s.enhancedSpan.SetStatus(enhanced.StatusCodeOk, "success")
+		}
 	}
 }
 
-// SetAttributes sets kv as attributes of the span.
+// SetAttributes sets attributes on both OTEL and enhanced spans.
 func (s *Span) SetAttributes(kv ...attribute.KeyValue) {
-	s.otelSpan.SetAttributes(kv...)
+	if s.otelSpan != nil {
+		s.otelSpan.SetAttributes(kv...)
+	}
+	if s.enhancedSpan != nil {
+		enhancedAttrs := convertToEnhancedAttributes(kv)
+		s.enhancedSpan.SetAttributes(enhancedAttrs...)
+	}
+}
+
+// SetAttribute sets a single attribute on both OTEL and enhanced spans.
+func (s *Span) SetAttribute(key string, value interface{}) {
+	if s.otelSpan != nil {
+		s.otelSpan.SetAttributes(Attribute(key, value))
+	}
+	if s.enhancedSpan != nil {
+		s.enhancedSpan.SetAttribute(key, value)
+	}
 }
 
 const spanDelimiter = "."
 
-// Name sets the span name by joining a list of strings in dot separated format.
-func Name(names ...string) string {
-	return strings.Join(names, spanDelimiter)
+// Name composes a span operation name using the standard delimiter.
+func Name(parts ...string) string {
+	return strings.Join(parts, spanDelimiter)
 }
 
-// Attribute takes a key value pair and returns attribute.KeyValue type.
+// SpanOperation returns a composite span operation name.
+func SpanOperation(parts ...string) string {
+	if len(parts) == 0 {
+		return ""
+	}
+	return strings.Join(parts, spanDelimiter)
+}
+
+// Attribute returns generic attribute.KeyValue type for any supported input types.
 func Attribute(k string, v any) attribute.KeyValue {
-	return keyValue(k, v)
+	switch v := v.(type) {
+	case bool:
+		return attribute.Bool(k, v)
+	case string:
+		return attribute.String(k, v)
+	case int:
+		return attribute.Int(k, v)
+	case int64:
+		return attribute.Int64(k, v)
+	case float64:
+		return attribute.Float64(k, v)
+	case []string:
+		return attribute.StringSlice(k, v)
+	}
+	return attribute.String(k, "<unsupported>")
 }
 
-// HTTPStatusCodeAttributes generates attributes of the HTTP namespace as specified by the OpenTelemetry
-// specification for a span.
+// HTTPStatusCodeAttributes returns the attributes list for the given status code.
 func HTTPStatusCodeAttributes(code int) []attribute.KeyValue {
 	return []attribute.KeyValue{semconv.HTTPStatusCodeKey.Int(code)}
 }
+
+// HTTPClient wraps the given RoundTripper and returns an http.Client with tracing instrumentation.
+// The given attributes will be part of the span attributes.
+func HTTPClient(rt http.RoundTripper, attrs ...attribute.KeyValue) *http.Client {
+	return &http.Client{
+		Transport: otelhttp.NewTransport(rt,
+			otelhttp.WithSpanOptions(trace.WithAttributes(attrs...)),
+		),
+	}
+}
+
+// NewHTTPClient is a convenience helper that returns a client with otel transport.
+func NewHTTPClient(base *http.Client) *http.Client {
```

Comment:
is this function used anywhere? I don't see any spot about it. please remove this